CONTRIBUTING.md

```diff
@@ -1,101 +1,125 @@
-# Welcome to the `esp-hal` contributing guide
 
-Thank you for investing your time in contributing to our project!
+Welcome to the `esp-hal` Contributing Guide
+===========================================
```

Comment:
Can we please use hashes for headers, all of our other `README.md` files use this convention.

Reply:
Oops, sorry, bad habit, will be fixed